### PR TITLE
Add indentation based on level and set correct collapsed state icon as per the design

### PR DESCRIPTION
**Pull request by MayaKirova**
Minor UI changes.

Note: Indentation size depends on display density, which will probably be smaller in the pivot grid.

Closes #  

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by skrustev** [comment]
I am not sure but for a configuration like the one bellow, it appears that the first row dimension receives and extra indentation for the first rows of each `All Products`,  while I think there should be no indentation at all for the first dimension:

UpperLevelIndentation

Configuration:


```typescript
        rows: [
            {
                memberName: 'City',
                enabled: true
            },
            {
                memberFunction: () => 'All',
                memberName: 'AllProducts',
                enabled: true,
                childLevel:
                {
                    memberFunction: (data) => data.ProductCategory,
                    memberName: 'ProductCategory',
                    enabled: true,
                }
            }

        ]
```

**Comment by skrustev** [comment]
1) For column dimensions it seems to apply the width specified to each value column. 
For example if there is column dimension with width 200px specified and there are two values in the config, the column dimension actually has 400px width.
It can be logged separately though

2) When moving column dimension chip to row dimension area the grid does not reflow. This is not reproduced in the pivot master.
PivotGrid

File: projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-row.component.html [hunk 1]
```diff
@@ -98,15 +98,25 @@
...
+            <igx-icon [attr.draggable]="false"
+                      (click)="grid.toggleRow(getRowDimensionKey(column))">
+                      {{ getExpandState(column) ? 'expand_more' : 'chevron_right'}}</igx-icon>
             {{column.header}}
         </div>
     </ng-template>
 
     <ng-template #headerTemplateGray let-column>
-        <div class="igx-grid__tr--header">
+        <div class='igx-grid__tr--header igx-grid__row-indentation--level-{{getLevel(column)}}'>
```
Comment (mddragnev) [inline projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-row.component.html:110]: Do we still need both default template and gray template ?

**Review by mddragnev — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-row.component.html (hunk 1 above)
Comment (dkamburov) [reply projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-row.component.html:110]: I guess this comment is for @desig9stein and @StefanIvanov. Should we drop completely gray styles before having the cell merging feature?

Comment (StefanIvanov) [reply projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-row.component.html:110]: We can drop it for now, this version is just a preview, so let's not add things that we wouldn't like to be there later on if this is the case here :)

Comment (MayaKirova) [reply projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-row.component.html:110]: Ok. In that case I'll remove the whole template.

**Review by mddragnev — APPROVED** [review]

**Pull request merged**